### PR TITLE
Switch to colon notation for dependencies.

### PR DESCRIPTION
**Pull request by anuraaga**
- Less verbose
- Format provided by search.maven.org and official docs in most places I've seen
- Consistent with `opentelemetry-java`

### CODE REVIEW

File: examples/distro/build.gradle [hunk 1]
```diff
@@ -17,12 +17,12 @@ subprojects {
...
-      bytebuddyagent      : dependencies.create(group: 'net.bytebuddy', name: 'byte-buddy-agent', version: versions.bytebuddy),
+      bytebuddy           : dependencies.create("net.bytebuddy:byte-buddy:${versions.bytebuddy}"),
+      bytebuddyagent      : dependencies.create("net.bytebuddy:byte-buddy-agent:${versions.bytebuddy}"),
       autoservice         : [
-        dependencies.create(group: 'com.google.auto.service', name: 'auto-service', version: '1.0-rc7'),
-        dependencies.create(group: 'com.google.auto', name: 'auto-common', version: '0.8'),
-        dependencies.create(group: 'com.google.guava', name: 'guava', version: "${versions.guava}"),
+        dependencies.create("com.google.auto.service:auto-service:1.0-rc7"),
+        dependencies.create("com.google.auto:auto-common:0.8"),
+        dependencies.create("com.google.guava:guava:${versions.guava}"),
```
Comment (mateuszrzeszutek) [inline examples/distro/build.gradle:25]: Nit: I think that we don't need to call `dependencies.create()` anymore, we can just pass the string.

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by iNikem — APPROVED** [review]

File: examples/distro/build.gradle (hunk 1 above)
Comment (anuraaga) [reply examples/distro/build.gradle:25]: Yay! Thanks!

File: buildSrc/src/main/groovy/MuzzlePlugin.groovy [hunk 2]
```diff
@@ -324,7 +324,7 @@ class MuzzlePlugin implements Plugin<Project> {
     def config = instrumentationProject.configurations.create(taskName)
 
     if (!muzzleDirective.coreJdk) {
-      def dep = instrumentationProject.dependencies.create("$versionArtifact.groupId:$versionArtifact.artifactId:$versionArtifact.version") {
```
Comment (mateuszrzeszutek) [inline buildSrc/src/main/groovy/MuzzlePlugin.groovy:327]: Wait, you probably need those in this file 😄

Comment (anuraaga) [reply buildSrc/src/main/groovy/MuzzlePlugin.groovy:327]: Yikes bad regex :)

**Pull request merged**